### ISSUE TITLE
Nightly failures on multiple unit tests

### ISSUE THREAD

**Issue by PrajwolAmatya** [owncloud/web#10380]
## Description
Multiple unit tests failed with the following log:

```
FAIL packages/web-app-preview/tests/unit/components/MediaControls.spec.ts
  ● Test suite failed to run

    A jest worker process (pid=138) was terminated by another process: signal=SIGKILL, exitCode=null. Operating system logs may contain more information on why this occurred.

      at ChildProcessWorker._onExit (node_modules/.pnpm/jest-worker@29.7.0/node_modules/jest-worker/build/workers/ChildProcessWorker.js:370:23)

FAIL packages/web-app-admin-settings/tests/unit/views/Groups.spec.ts
  ● Test suite failed to run

    A jest worker process (pid=125) was terminated by another process: signal=SIGKILL, exitCode=null. Operating system logs may contain more information on why this occurred.

      at ChildProcessWorker._onExit (node_modules/.pnpm/jest-worker@29.7.0/node_modules/jest-worker/build/workers/ChildProcessWorker.js:370:23)
```

Builds:
- https://drone.owncloud.com/owncloud/web/42019/8/4
- https://drone.owncloud.com/owncloud/web/42021/8/4

**Comment by Salipa-Gurung** [comment]
Unit tests failed with log:
```
FAIL packages/web-app-files/tests/unit/views/spaces/GenericTrash.spec.ts
  ● Test suite failed to run

    A jest worker process (pid=125) was terminated by another process: signal=SIGKILL, exitCode=null. Operating system logs may contain more information on why this occurred.

      at ChildProcessWorker._onExit (node_modules/.pnpm/jest-worker@29.7.0/node_modules/jest-worker/build/workers/ChildProcessWorker.js:370:23)
```
Build: https://drone.owncloud.com/owncloud/web/42119/8/4

**Comment by kulmann** [comment]
The actual issue is stated above the failures:

```
A worker process has failed to exit gracefully and has been force exited. This is likely caused by tests leaking due to improper teardown. Try running with --detectOpenHandles to find leaks. Active timers can also cause this, ensure that .unref() was called on them.
```

Could you try to run the unit tests with the `--detectOpenHandles` option and see if it detects anything?

**Comment by JammingBen** [comment]
Some more insight from my side (basically copied from chat):

When we first stumbled upon this issue a few months ago, limiting the `maxWorkers` to 50% helped. But the pipeline runs with this setting as well, so apparently it doesn't help anymore.

I tried debugging the issue back then with `--detectOpenHandles` but had no success as it didn't detect anything.

The failing tests seem to be different each run, so to me it looks like a general problem with our test suite, or individual tests leaking. Or maybe related to the circular import problems we oft have with our unit tests. Jest seems to be pretty bad at resolving things... vite does it just fine (btw @dschmidt pointed out a few days ago that `vitest` might be worth a look for the future).
